lochness/mindlamp/__init__.py

```diff
@@ -38,46 +92,95 @@ def sync(Lochness: 'lochness.config',
-        dst = os.path.join(
-                dst_folder,
-                f'{subject_id}_{subject.study}_{data_name}.json')
-
-        jsonData = json.dumps(
-            data_dict,
-            sort_keys=True, indent='\t', separators=(',', ': '))
-
-        content = jsonData.encode()
-
-        if not Path(dst).is_file():
-            lochness.atomic_write(dst, content)
-        else:  # compare existing json to the new json
-            crc_src = lochness.crc32(content.decode('utf-8'))
-            crc_dst = lochness.crc32file(dst)
-            if crc_dst != crc_src:
-                logger.warn(f'file has changed {dst}')
-                lochness.backup(dst)
-                logger.debug(f'saving {dst}')
+    # how many days of data from current time, default past 10 days
+    days_to_check = get_days_to_pull(Lochness)
+
+    # current time (ct) in UTC
+    ct_utc = datetime.now(pytz.timezone('UTC'))
+
+    # set the cut off point as UTC 00:00:00.00
+    ct_utc_00 = ct_utc.replace(hour=0, minute=0, second=0, microsecond=0)
+
+    for days_from_ct in range(days_to_check):
+        # n day before current time
+        time_utc_00 = ct_utc_00 - timedelta(days=days_from_ct)
+        time_utc_00_ts = time.mktime(time_utc_00.timetuple()) * 1000
+        time_utc_24 = time_utc_00 + timedelta(hours=24)
+        time_utc_24_ts = time.mktime(time_utc_24.timetuple()) * 1000
+
+        # date string to be used in the file name
+        date_str = time_utc_00.strftime("%Y_%m_%d")
+
+        # Extra information for future version
+        # study_id, study_name = get_study_lamp(LAMP)
+        # subject_ids = get_participants_lamp(LAMP, study_id)
+        subject_id = subject.mindlamp[f'mindlamp.{subject.study}'][0]
+        logger.debug(f'Mindlamp {subject_id} {date_str} data pull - start')
+
+        # pull data from mindlamp
+        begin = time.time()
+        activity_dicts = get_activity_events_lamp(
+                LAMP, subject_id,
+                from_ts=time_utc_00_ts, to_ts=time_utc_24_ts)
+
+        sensor_dicts = get_sensor_events_lamp(
+                LAMP, subject_id,
+                from_ts=time_utc_00_ts, to_ts=time_utc_24_ts)
+        end = time.time()
+        logger.debug(f'Mindlamp {subject_id} {date_str} data pull - complete')
+
+        # set destination folder
+        # dst_folder = tree.get('mindlamp', subject.general_folder)
+        dst_folder = tree.get('mindlamp',
+                              subject.protected_folder,
+                              processed=False,
+                              BIDS=Lochness['BIDS'])
+
+        # store both data types
+        for data_name, data_dict in zip(['activity', 'sensor'],
+                                        [activity_dicts, sensor_dicts]):
+
+            dst = os.path.join(
+                    dst_folder,
+                    f'{subject_id}_{subject.study}_{data_name}_'
+                    f'{date_str}.json')
```

Comment:
Isn't this becoming a long file name? How about at least dorpping `subject.study` since it is already residing in `subject.study` folder?

Reply:
I think long names may come in handy in these multi-site studies like U24.  I insist keep a detailed file name for this study if it does not affect the following processing pipelines. We can bring this up on Monday meeting if needed.